Pod/Classes/ViewControllers/ARMapContainerViewController.m

```diff
@@ -1,26 +1,55 @@
 #import "ARMapContainerViewController.h"
 #import "ARMapComponentViewController.h"
 #import "ARCityComponentViewController.h"
+#import "ARCityPickerComponentViewController.h"
 #import "ARComponentViewController.h"
+#import "ARCity.h"
 
 #import <FLKAutoLayout/UIView+FLKAutoLayout.h>
 
 @import Pulley;
+@import CoreLocation;
 
-@interface ARMapContainerViewController () <PulleyDelegate, PulleyDrawerViewControllerDelegate>
+@interface ARMapContainerViewController () <PulleyDelegate, PulleyDrawerViewControllerDelegate, CLLocationManagerDelegate>
 
 @property (nonatomic, readwrite) PulleyViewController *bottomSheetVC;
 @property (nonatomic, readwrite) ARMapComponentViewController *mapVC;
 @property (nonatomic, readwrite) ARCityComponentViewController *cityVC;
+@property (nonatomic, strong) CLLocationManager *locationManager;
+
+@property (nonatomic, strong) ARCityPickerComponentViewController *cityPickerController;
+@property (nonatomic, strong) UIView *cityPickerContainerView; // Need a view to have its own shadow.
 
 @end
 
+/*
+This is the top-level Local Discovery component, and should therefore be responsible for checking a user's location and
+displaying appropriate UI (did they accept the permissions prompt? are they need a city?). There are two cases:
+- Show the city picker (they didn't accept the prompt, or they're far away from a city).
+- Show the mapVC and cityVC with the appropriate city.
+Since this controller already has to do the above logic, having it handle the CityPicker interactions makes sense.
+*/
+
 @implementation ARMapContainerViewController
 
 - (void)viewDidLoad
 {
     [super viewDidLoad];
 
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.delegate = self;
+    self.locationManager.desiredAccuracy = kCLLocationAccuracyThreeKilometers;
+    [self.locationManager requestWhenInUseAuthorization];
+
+    __weak typeof(self) sself = self;
+    [[NSNotificationCenter defaultCenter] addObserverForName:@"ARLocalDiscoveryOpenCityPicker" object:nil queue:nil usingBlock:^(NSNotification * _Nonnull note) {
```

Comment:
Only question is why are we using `NSNotificationCenter` if events will only be emitted from this class and child React Native view controllers. Can't we just rely on communication channels between native and react native. 

Calling an Objective-C method from Javascript: https://facebook.github.io/react-native/docs/native-modules-ios#content

Reply:
Unfortunately, our setup doesn’t make this easy (how would we call a method on this specific controller instance?). I confirmed on Slack that this is how we’d want to approach this (especially because of imminent RN bridge changes): https://artsy.slack.com/archives/C02BAQ5K7/p1551463932014200

Reply:
Right yeah that makes, very true the RN bridge API will change once we upgrade as well. Sounds good